assets/js/modules/modal.js

```diff
@@ -0,0 +1,116 @@
+ *
+ * @see https://www.npmjs.com/package/a11y-dialog
+ */
+import A11yDialog from 'a11y-dialog'
+
+/**
+ * Default cfg
+ * @param {string}  containerId  (Required) Desired id of the container
+ * @param {array} containerClasses  (optional) Desired classes of the container
+ * @param {array} contentClasses  (optional)  Desired classes of the actual modal
+ * @param {string}  content (Required) Content of the modal
+ * @param {DOM node}  targetToggleHidden  (optional) Elements hidden when the modal is open
+ */
+const defaultConfig = {
+  containerId: '',
+  containerClasses: [],
+  contentClasses: [],
+  content: '',
+  targetToggleHidden: document.querySelectorAll('body > *:not(.modal):not([aria-hidden="true"]):not(script)'),
+}
+
+function Modal(cfg) {
+  // Merge default with current cfg
+  this.config = Object.assign({}, defaultConfig, cfg)
+  this.build()
+}
+
+Modal.prototype.open = function open() {
+  if (!this.dialog) {
+    console.error('Cannot open Modal, it does not exist')
+
+    return
+  }
+  this.dialog.show()
+}
+
+Modal.prototype.close = function close() {
+  if (!this.dialog) {
+    console.error('Cannot close Modal, it does not exist')
+
+    return
+  }
+  this.dialog.hide()
+}
+
+Modal.prototype.destroy = function destroy() {
+  if (!this.dialog) {
+    console.error('Cannot destroy Modal, it does not exist')
+
+    return
+  }
+  if (this.container) {
+    this.container.remove()
+  }
+  this.dialog.destroy()
+}
+
+Modal.prototype.init = function init() {
+  if (this.dialog) {
+    console.error('The modal is already init')
+
+    return
+  }
+  this.dialog = new A11yDialog(this.container, this.config.targetToggleHidden)
+}
+
+Modal.prototype.build = function build() {
+  // Creates the container
+  this.container = document.createElement('div')
+  this.container.id = this.config.containerId
+  this.container.classList.add('modal', ...this.config.containerClasses)
+  this.container.setAttribute('aria-hidden', true)
+
+  // Creates the overlay
+  const overlay = document.createElement('div')
+  overlay.classList.add('modal-overlay')
+  overlay.setAttribute('data-a11y-dialog-hide', true)
+  this.container.appendChild(overlay)
+
+  // Creates the close button
+  const btnClose = document.createElement('button', { type: 'button' })
+  btnClose.classList.add('modal-closebtn', 'btnIcon')
+  btnClose.setAttribute('data-a11y-dialog-hide', true)
+  btnClose.innerHTML = '<svg height="12" width="12"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#symbol-cross"></use></svg>'
+
+  // Creates the modal & fill it w/ the @param: content
+  const content = document.createElement('div')
+  content.classList.add('modal-content', ...this.config.contentClasses)
+  content.setAttribute('role', 'dialog')
+  content.innerHTML = this.config.content
+  content.appendChild(btnClose)
+
+  return this.container.appendChild(content)
+}
+
+Modal.prototype.insert = function insert() {
+  if (document.body.contains(this.container)) {
```

Comment:
We should use a boolean to know it and not traverse all the DOM tree.
`this.isInserted = true` when inserted.
`this.isInserted = false` when new modal are created (constructor)